### ISSUE TITLE
Traceback with Cylc 8 workflows

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#4315]
Reverse of #4309

Add logic to Cylc 7 to more gracefully handle Cylc 8 workflows.

Example traceback:

```
int() argument must be a string or a number, not 'NoneType'
```

Also ensure that the Cylc 7 logic for removing the contact files of crashed suites (included in `cylc scan` etc) does not interfere with Cylc 8 workflows.

**Comment by oliver-sanders** [comment]
> **See also** the related Rose issue: https://github.com/metomi/rose/issues/2474

**Comment by oliver-sanders** [comment]
Closed by #4320
